holidays/countries/nepal.py

```diff
@@ -0,0 +1,304 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.calendars.gregorian import JAN, FEB, MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT, NOV, DEC
+from holidays.constants import PUBLIC, WORKDAY
+from holidays.groups import (
+    ChristianHolidays,
+    HinduCalendarHolidays,
+    InternationalHolidays,
+    IslamicHolidays,
+    StaticHolidays,
+)
+from holidays.holiday_base import HolidayBase
+
+
+class Nepal(
+    HolidayBase,
+    ChristianHolidays,
+    HinduCalendarHolidays,
+    InternationalHolidays,
+    IslamicHolidays,
+    StaticHolidays,
+):
+    """Nepal holidays.
+
+    References:
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Nepal>
+        * <https://narayanilawfirm.org.np/list-of-public-holidays-in-nepal-2079/>
+        * <https://www.timeanddate.com/holidays/nepal/>
+    """
+
+    country = "NP"
+    supported_categories = (PUBLIC, WORKDAY)
+    # %s (estimated).
+    estimated_label = "%s (estimated)"
+    start_year = 2010
+
+    def __init__(self, islamic_show_estimated: bool = True, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        HinduCalendarHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(
+            self, cls=NepalIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        StaticHolidays.__init__(self, cls=NepalStaticHolidays)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        if self._year >= 2023:
+            # Prithvi Jayanti.
+            self._add_holiday_jan_11("Prithvi Jayanti")
+
+        if self._year not in {2019, 2020}:
+            # Martyr's Day.
+            self._add_holiday_jan_30("Martyr's Day")
+
+        if self._year not in {2019, 2020}:
+            # National Democracy Day.
+            self._add_holiday_feb_19("National Democracy Day")
```

Comment:
Maybe considering putting the holidays that was cut in 2019-2020 in the same section - maybe at the end of `_populate_public_holidays` to reduce the number of duplicate year checks down a bit:
```suggestion
        # Removed by MoHA between 2019-2020.
        if self._year not in {2019, 2020}:
            # Martyr's Day.
            self._add_holiday_jan_30("Martyr's Day")

            # National Democracy Day.
            self._add_holiday_feb_19("National Democracy Day")

            # Republic Day.
            self._add_holiday_may_29("Republic Day")
            .
            .
            .
```